### PR TITLE
Document explicitly checking type property when using forEach

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

Ref https://github.com/stylelint/stylelint/pull/4744#discussion_r420705135

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: docs/developer-guide/rules.md [hunk 1]
```diff
@@ -25,7 +25,15 @@ You should make use of the:
 
 #### PostCSS API
 
-Use the [PostCSS API](https://api.postcss.org/) to navigate and analyze the CSS syntax tree. We recommend using the `walk` iterators (e.g. `walkDecls`), rather than using `forEach` to loop through the nodes. Use `node.raws` instead of `node.raw()` when accessing raw strings from the [PostCSS AST](https://astexplorer.net/#/gist/ef718daf3e03f1d200b03dc5a550ec60/c8cbe9c6809a85894cebf3fb66de46215c377f1a).
+Use the [PostCSS API](https://api.postcss.org/) to navigate and analyze the CSS syntax tree. We recommend using the `walk` iterators (e.g. `walkDecls`), rather than using `forEach` to loop through the nodes. If `forEach` is a better fit, you should explicitly check the `type` property of the `node` before attempting to access other properties. For example:
```
Comment (m-allanson) [inline docs/developer-guide/rules.md:28]: How can I identify when `forEach` would be a better fit?

**Comment by m-allanson** [comment]
Looks good! One comment but I think this is good to merge regardless.

File: docs/developer-guide/rules.md (hunk 1 above)
Comment (jeddy3) [reply docs/developer-guide/rules.md]: Good question. It highlights that `forEach` is never a good fit.

I've amended the pull request to clarify:

- that checking the type is for when an array method is used
- it is quite separate from the `forEach` advice

Comment (m-allanson) [reply docs/developer-guide/rules.md]: 👍

**Pull request merged**